packages/core/admin/ee/server/seat-enforcement.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+const _ = require('lodash');
+
+const enableUsersToLicenseLimit = async (numberOfUsersToEnable) => {
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (!data || !data.value || data.value.length === 0) return;
+
+  const disabledUsers = JSON.parse(data.value);
+
+  const usersToEnable = _.take(disabledUsers, numberOfUsersToEnable);
+
+  usersToEnable.forEach(async (user) => {
```

Comment:
Is it supposed to be run async ? What if an error is thrown after the function has finished?
You could use `for await (const ... of ...)` instead, to make sure everything is run in sequence, or `Promise.all(usersToEnabled.map(x => query())` to ensure everything is run in parrallel.

Reply:
For context:
![image](https://user-images.githubusercontent.com/25851739/218700199-c04c5c40-4322-4f50-a487-8e4d65ecf04a.png)
![image](https://user-images.githubusercontent.com/25851739/218700226-a36ff9a0-63d5-4f53-9d76-d9985ee08742.png)

Reply:
And with map/promise.all (would be the same with `for await of`, except it would be run in sequence)
![image](https://user-images.githubusercontent.com/25851739/218700876-2bb40bfe-d459-4b60-b9f7-b040d1d4d5c2.png)
![image](https://user-images.githubusercontent.com/25851739/218700946-99327467-3e9e-41a1-a498-1a7055cf49bd.png)

Reply:
Ooooh, this is a really good catch, I completely didn't think about this. Thank you! 👍